common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1859,8 +1861,14 @@ def __init__(
         self.descriptor_runtime = descriptor_runtime
         self.rebind_noauth_module_to_user = rebind_noauth_module_to_user
 
-        if user:
-            self.user_id = user.id
+        # Fetch user data from the user service
+        user_service = self.service(self, 'user')
+        self.user_id = user_service.get_current_user_id() if user_service else None
```

Comment:
Previously, the `user_id` attribute was only set if `user` was non-null. Is this an issue? (see above re CombinedSystem)